libfreerdp/core/settings.c

```diff
@@ -252,13 +252,27 @@ void settings_load_hkey_local_machine(rdpSettings* settings)
 
 BOOL settings_get_computer_name(rdpSettings* settings)
 {
-	DWORD nSize = MAX_COMPUTERNAME_LENGTH + 1;
-	CHAR computerName[MAX_COMPUTERNAME_LENGTH + 1];
+	DWORD nSize = 0;
+	CHAR* computerName;
+
+	if (GetComputerNameExA(ComputerNameNetBIOS, NULL, &nSize))
```

Comment:
Here and below: The actual fix must include the GetLastError() check and eventually a size check.
Something like this:
```
if (ComputerNameNetBIOS, NULL, &nSize) || GetLastError() != ERROR_MORE_DATA ||  nSize < 2)
    return FALSE;
```